ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -77,17 +79,84 @@ class Endpoints(
   private def toRoute(res: => Future[Error \/ SearchResult[Error \/ JsValue]]): Route =
     responseToRoute(httpResponse(res))
 
+  // Always put this directive after a path to ensure
+  // that you don't log request bodies multiple times (simply because a matching test was made multiple times).
+  // TL;DR JUST PUT THIS THING AFTER YOUR FINAL PATH MATCHING
```

Comment:
Just trying to clarify here that it is considered to be a bug if the directive is being executed without the path afterwards matching. This should prevent us from printing request bodies which are targeted to unmatched routes (And also as described in the comment can result in printing the body more than once).

Reply:
I guess this is the drawback of the directives, one has to be quite careful using them.